test/services/onlinetesting/OnlineTestServiceSpec.scala

```diff
@@ -218,88 +272,87 @@ class OnlineTestServiceSpec extends PlaySpec with BeforeAndAfterEach with Mockit
           .thenReturn(Future.successful(registration))
-        when(cubiksGatewayClientMock.inviteApplicant(eqTo(inviteApplicant))(any[HeaderCarrier]))
+        when(cubiksGatewayClientMock.inviteApplicant(any[InviteApplicant])(any[HeaderCarrier]))
           .thenReturn(Future.successful(invitation))
         when(otRepositoryMock.storeOnlineTestProfileAndUpdateStatusToInvite(ApplicationId, onlineTestProfile))
           .thenReturn(Future.successful(()))
         when(cdRepositoryMock.find(UserId)).thenReturn(Future.successful(contactDetails))
         when(emailClientMock.sendOnlineTestInvitation(eqTo(EmailContactDetails), eqTo(PreferredName), eqTo(ExpirationDate))(
           any[HeaderCarrier]
         )).thenReturn(Future.successful(()))
+
         when(otRepositoryMock.storeOnlineTestProfileAndUpdateStatusToInvite(ApplicationId, onlineTestProfile))
           .thenReturn(Future.failed(new Exception))
-        when(otprRepositoryMock.remove(ApplicationId)).thenReturn(Future.successful(()))
         when(trRepositoryMock.remove(ApplicationId)).thenReturn(Future.successful(()))
 
-        val result = onlineTestService.registerAndInviteApplicant(applicationForOnlineTestingWithNoTimeAdjustments)
+        val result = onlineTestService.registerAndInviteForTestGroup(applicationForOnlineTestingWithNoTimeAdjustments)
         result.failed.futureValue mustBe an[Exception]
 
-        verify(emailClientMock).sendOnlineTestInvitation(eqTo(EmailContactDetails), eqTo(PreferredName), eqTo(ExpirationDate))(
+        verify(emailClientMock, times(0)).sendOnlineTestInvitation(any[String], any[String], any[DateTime])(
           any[HeaderCarrier]
         )
-        verify(auditServiceMock).logEventNoRequest("UserRegisteredForOnlineTest", auditDetails)
-        verify(auditServiceMock).logEventNoRequest("UserInvitedToOnlineTest", auditDetails)
-        verify(auditServiceMock).logEventNoRequest("OnlineTestInvitationEmailSent", auditDetailsWithEmail)
-        verify(auditServiceMock, times(3)).logEventNoRequest(any[String], any[Map[String, String]])
+        verify(auditServiceMock, times(2)).logEventNoRequest("UserRegisteredForOnlineTest", auditDetails)
+        verify(auditServiceMock, times(2)).logEventNoRequest("UserInvitedToOnlineTest", auditDetails)
+        verify(auditServiceMock, times(0)).logEventNoRequest("OnlineTestInvitationEmailSent", auditDetailsWithEmail)
+        verify(auditServiceMock, times(4)).logEventNoRequest(any[String], any[Map[String, String]])
       }
     "audit 'OnlineTestInvitationProcessComplete' on success" in new OnlineTest {
       when(cubiksGatewayClientMock.registerApplicant(eqTo(registerApplicant))(any[HeaderCarrier]))
         .thenReturn(Future.successful(registration))
-      when(cubiksGatewayClientMock.inviteApplicant(eqTo(inviteApplicant))(any[HeaderCarrier]))
+      when(cubiksGatewayClientMock.inviteApplicant(any[InviteApplicant])(any[HeaderCarrier]))
```

Comment:
Generally speaking, it should work with inviteApplication or registerApplicant. It used to work at least.